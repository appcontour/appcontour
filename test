#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
