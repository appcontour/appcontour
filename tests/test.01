./contour applyrule n1 <examples/tre_sfere.morse
