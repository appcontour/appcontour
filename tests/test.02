#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
