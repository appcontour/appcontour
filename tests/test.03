./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
