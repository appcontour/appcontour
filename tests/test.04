./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
