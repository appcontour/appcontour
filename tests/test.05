./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
