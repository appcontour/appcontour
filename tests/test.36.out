Example knots/knot_3_1.knot:
<a,b; aaBAbAB>
(1,1): +1+a-aaBA-aaBAbA
(1,2): -aaB+aaBA-aaBAbAB
(1,1): +1-t+t^2
(1,2): 0
Example links/link_2_6_3.knot:
<a,b,c; abCacAcBAC, ababaCAcABAc>
(1,1): +1+abC-abCacA-abCacAcBA
(1,2): +a-abCacAcB
(1,3): -abC+abCa+abCacA-abCacAcBAC
(2,1): +1+ab+abab-ababaCA-ababaCAcA-ababaCAcABA
(2,2): +a+aba-ababaCAcAB
(2,3): -ababaC+ababaCA+ababaCAcABA
(1,1): -uv^(-1)+u^2v^(-1)+u
(1,2): -u^2v^(-3)-uv^(-2)+u^2v^(-2)-v^(-1)+uv^(-1)+1
(1,3): +u^2v^(-3)-u^3v^(-3)+uv^(-2)-u^2v^(-2)+v^(-1)-uv^(-1)
(2,1): +u-uv
(2,2): +uv^(-2)-uv^(-1)+1-v
(2,3): -uv^(-2)+u^2v^(-2)-1+u
Example borromeanrings.knot:
<a,b,c; acACbcaCAB, acAbaBCbAB>
(1,1): +1-acA+acACbc-acACbcaCA
(1,2): +acAC-acACbcaCAB
(1,3): +a-acAC+acACb-acACbcaC
(2,1): +1-acA+acAb-acAbaBCbA
(2,2): +acA-acAbaB+acAbaBC-acAbaBCbAB
(2,3): +a-acAbaBC
(1,1): +1-v-w+vw
(1,2): 0
(1,3): -1+u+v-uv
(2,1): +1-v-w+vw
(2,2): -1+u+w-uw
(2,3): 0
Example embrace.knot:
<a,b,c; abABcAC>
(1,1): +1-abA-abABcA
(1,2): +a-abAB
(1,3): +abAB-abABcAC
(1,1): -1+u+v
(1,2): 0
(1,3): 0
